### ISSUE TITLE
How to update refresh_token in source config?

### ISSUE THREAD

**Issue by anna-lisniak** [faros-ai/airbyte-connectors#1019]
My source needs to support OAuth2.0 authentication.
This gives me `access_token`, `refresh_token`, `expires_in`
So when the acess_token is expired, I have to request a new access_token and a new refresh_token. 
Do you know how I can update the new tokens in the configuration?

**Comment by tovbinm** [comment]
Which source are you referring to?

**Comment by Vadym5** [comment]
We have the same issue. We have refresh_token which updates at some time. And we need to update it in the connector source. Can we do it?

**Comment by tovbinm** [comment]
Which connector are you using?

**Comment by Vadym5** [comment]
> Which connector are you using?

Custom connector for Quickbooks service.

**Comment by tovbinm** [comment]
Yes, you should be able to do it. I recommend implementing the client in a separate class and explicitly handling the token expiration error in the client.
